nautobot/extras/jobs.py

```diff
@@ -163,7 +165,11 @@ def __call__(self, *args, **kwargs):
     def __str__(self):
         return str(self.name)
 
-    # See https://github.com/PyCQA/pylint-django/issues/240 for why we have a pylint disable on each classproperty below
+    def fail(self, msg, *args, **kwargs):
+        """Mark this job as failed without immediately raising an exception and aborting."""
+        # Instead of failing in "fail" grouping, fail in the parent function's grouping by default
```

Comment:
Does this have to do with the stacklevel = 2 declaration below?

Reply:
Yes, same idea as the other comment above. Without it, the failure gets reported as occurring within `fail()` instead of within the function that *called* `fail()`.

Reply:
before:
<img width="1096" alt="image" src="https://github.com/user-attachments/assets/b079a777-d431-4f55-b322-7fd8d6ffa67d" />

after:
<img width="1090" alt="image" src="https://github.com/user-attachments/assets/4560d710-aeb3-46cb-b04e-fa28fc24ed7d" />